core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -336,9 +336,9 @@ private[scheduler] object BlacklistTracker extends Logging {
    */
   def isBlacklistEnabled(conf: SparkConf): Boolean = {
     conf.get(config.BLACKLIST_ENABLED) match {
-      case Some(enabled) =>
+      case enabled =>
         enabled
-      case None =>
+      case _ =>
```

Comment:
On the other hand, we could update the doc to change to "(none)" as default instead of "false".